pkg/manager/manager_test.go

```diff
@@ -1044,6 +1045,54 @@ var _ = Describe("manger.Manager", func() {
 				}))).NotTo(Succeed())
 			})
 
+			It("should not return runnables context.Canceled errors", func() {
+				Expect(options.Logger).To(BeZero(), "this test overrides Logger")
+
+				var log struct {
+					sync.Mutex
+					messages []string
+				}
+				options.Logger = funcr.NewJSON(func(object string) {
+					log.Lock()
+					log.messages = append(log.messages, object)
+					log.Unlock()
+				}, funcr.Options{})
+
+				m, err := New(cfg, options)
+				Expect(err).NotTo(HaveOccurred())
+				for _, cb := range callbacks {
+					cb(m)
+				}
+
+				// Runnables may return ctx.Err() as shown in some [context.Context] examples.
+				started := make(chan struct{})
+				Expect(m.Add(RunnableFunc(func(ctx context.Context) error {
+					close(started)
+					<-ctx.Done()
+					return ctx.Err()
+				}))).To(Succeed())
+
+				stopped := make(chan error)
+				ctx, cancel := context.WithCancel(context.Background())
+				go func() {
+					stopped <- m.Start(ctx)
+				}()
+
+				// Wait for runnables to start, signal the manager, and wait for it.
+				<-started
+				cancel()
+				Expect(<-stopped).To(Succeed())
```

Comment:
Q: Couldn't it be that l.448 reads and returns the error from errChan?

Reply:
Here, in this test? I don't think so.

I don't fully understand all the startup logic, but the runnable groups don't appear to send to that channel during Start (...exactly). The leader election goroutine does.

Reply:
Let me rephrase.

As far as I can tell there are two places reading from errChan:
```go
	select {
	case <-ctx.Done():
		// We are done
		return nil
	case err := <-cm.errChan:
		// Error starting or running a runnable
		return err
	}
```

```
			select {
			case err := <-cm.errChan:
				if !errors.Is(err, context.Canceled) {
					cm.logger.Error(err, "error received after stop sequence was engaged")
				}
			case <-stopComplete:
				return
			}
```

This test case verifies that the cancel error ends up in the logger and not as a return parameter of `stopped <- m.Start(ctx)`. I'm wondering if the cancel error always ends up in the logger, and if yes, why.

Reply:
Nevermind. engageStopProcedure is only called after start returns